src/basic/strv.c

```diff
@@ -936,6 +936,64 @@ static int string_strv_hashmap_put_internal(Hashmap *h, const char *key, const c
         return 1;
 }
 
+/* The equivalent of strv_parse_nulstr, but for sensitive data. */
+char **strv_parse_password(const char *s, size_t l) {
```

Comment:
This duplication is bad. It should be possible to adapt `strv_parse_nulstr` into `strv_parse_nulstr_full` that takes a flag (or function pointer...) to optionaly use erase_freep_and_unlock when appropriate.